tests/console/hibernation_check.pm

```diff
@@ -13,10 +13,15 @@ use base "consoletest";
 use strict;
 use warnings;
 use testapi;
+use Utils::Architectures;
 
 sub run {
     select_console 'root-console';
-    assert_script_run("grep -v 'resume=' /proc/cmdline", fail_message => "resume parameter found in /proc/cmdline");
+    if (is_x86_64) {
```

Comment:
we don't need `if/else` based on architectures, perhaps just as a comment to clarify that only happens in x86_64, it is better just to do the logic based on test data. You can add `resume_kernel_param_present: 1` and `resume_kernel_param_present: 0` in test data and then here you read that that and  you add in single line `$is_present ? `-v` : ''`, that would be less repetitive code.

Reply:
To add `resume_kernel_param_present: 1` and `resume_kernel_param_present: 0` to YAML_TEST_DATA? Now the x86_64, aarch64 and s390x_kvm are using the same YAML_TEST_DATA, if add resume_kernel_param_present to test data, then need to add one more YAML_TEST_DATA, is that fine?

Reply:
ok, good point, to unify yaml for test data could be something to think better how in the future, they change a lot among architectures, perhaps a better solution would be to extend the distribution pattern to accomplish different validation.

For now, to avoid that fragmentation, we can just polish a bit your current code, storing in a variable an empty string or the `-v` depending on the condition, you can use ternary, condition ?  value1 : value2 in interpolate in single assert_script_run, and we are done for now.

Reply:
forget my last paragraph, we already have different test data file for several cases, it would make sense to create an specific one for x86_64 and leave the other ones with current name.